src/cascadia/TerminalSettingsModel/Command.cpp

```diff
@@ -65,10 +68,140 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         return _subcommands ? _subcommands.GetView() : nullptr;
     }
 
+    // Function Description:
+    // - reports if the current command has nested commands
+    // - This CANNOT detect { "name": "foo", "commands": null }
     bool Command::HasNestedCommands() const
     {
         return _subcommands ? _subcommands.Size() > 0 : false;
     }
+
+    // Function Description:
+    // - reports if the current command IS a nested command
+    // - This CAN be used to detect cases like { "name": "foo", "commands": null }
+    bool Command::IsNestedCommand() const noexcept
+    {
+        return _nestedCommand;

```

Comment:
eehhhh? what?

Reply:
I hate every bit of this. The difference is...
- `HasNestedCommands`: do I contain any sub-commands
- `IsNestedCommand`: was I using the `"commands"` key in the JSON

If `HasNestedCommands` == `true`, then `IsNestedCommand` == `true`. But this is an IF, not an IFF.

So I'm using this as a way to detect `"commands": null` as a nested command that contains nothing.